x-pack/plugins/observability/public/application/index.tsx

```diff
@@ -30,8 +32,11 @@ function getTitleFromBreadCrumbs(breadcrumbs: Breadcrumbs) {
 }
 
 function App() {
+  const isLandingPage = useRouteMatch('/landing');
+
   return (
     <>
+      {!isLandingPage && <ManageHasDataFetches />}
```

Comment:
Is `ManageHasDataFetches` supposed to render anything? If not shouldn't it be a hook?

Reply:
Component is needed to manage hooks, hooks can't be called inside hook principle applies here.

Reply:
> hooks can't be called inside hook principle applies here

I'm not following? Hooks can be composed of other hooks.

Reply:
no you can't call a hook inside another hook.

Reply:
https://reactjs.org/docs/hooks-rules.html#only-call-hooks-from-react-functions

Reply:
You should call a hook from a React component but you can call a hook from another hook... that's what you're doing here https://github.com/elastic/kibana/pull/79388/files#diff-8eb08ee658d10265529084304f49a4deR19

Reply:
@shahzad31 From that very page :)

![image](https://user-images.githubusercontent.com/209966/95185112-99325b80-07c8-11eb-9722-33e1966829f4.png)

Reply:
i think a shared component is needed in any case to keep state, no? i guess a hook can be called from application/index.tsx component to get the same purpose...

Reply:
They didn't write it explicitly but you can also call custom hooks from custom hooks.

Reply:
Example of calling a composing custom hooks:

https://github.com/elastic/kibana/blob/08a4586b143204c350a2603e3cb92abb5459858a/x-pack/plugins/apm/public/hooks/useKibanaUrl.ts#L10-L19

Reply:
ahh ok, i am pretty packed for today, i have asked @cauemarcondes to pick up the rest of the refactor. if we want to make it to 7.10FF. otherwise PR is in pretty good shape. I have no objection if he can refactor this component to a hook.

Reply:
> i think a shared component is needed in any case to keep state, no? 

You can use `useState` for state. If you want to share it between several components you can lift the hook invocation to a shared parent component. Or you can use context.

Reply:
yes are right, i meant if you are calling hook multiple times, but yes if we call hook from parent hook, that will essentially act as what this component is doing.

Reply:
I used to use components this way but I think it's become an anti-pattern after the introduction of hooks. Components are specifically for rendering ui, whereas hooks are for calculating logic with side effects. imo this separation makes it much easier to understand the intention.

Reply:
totally agree.

Reply:
fixed here https://github.com/elastic/kibana/pull/79388/commits/7c6aa10e4fe925dfcd72d767b863f9da30ed78db